manipulation/station.py

```diff
@@ -265,6 +265,195 @@ def add_directives(
     return scenario
 
 
+def _FindChildren(flattened_directives, model_instance_names):
```

Comment:
What do you think about type hints for private functions/members? Typing could be helpful for others using that functionality in the future, but detailed docstrings seem less necessary.